test/one/profiler/test/TestDeclaration.java

```diff
@@ -0,0 +1,148 @@
+            }
+        }
+
+        List<String> allTestDirs = new ArrayList<>();
+        File[] files = new File("test/test").listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    allTestDirs.add(file.getName());
+                }
+            }
+        }
+
+        return new TestDeclaration(allTestDirs, filters, skipFilters);
+    }
+
+    private static List<RunnableTest> getRunnableTests(Class<?> cls) {
+        List<RunnableTest> rts = new ArrayList<>();
+        for (Method m : cls.getMethods()) {
+            for (Test t : m.getAnnotationsByType(Test.class)) {
+                rts.add(new RunnableTest(m, t));
+            }
+        }
+        return rts;
+    }
+
+    private static List<RunnableTest> getRunnableTests(String dir) {
+        String className = "test." + dir + "." + Character.toUpperCase(dir.charAt(0)) + dir.substring(1) + "Tests";
+        try {
+            return getRunnableTests(Class.forName(className));
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private List<RunnableTest> getRunnableTests(List<String> globs) {
+        return globs.stream().filter(g -> g.equals(g.toLowerCase())).flatMap(g -> getRunnableTests(g).stream()).sorted(Comparator.comparing(RunnableTest::testName)).collect(Collectors.toList());
+    }
+
+    public List<RunnableTest> getRunnableTests() {
+        return getRunnableTests(allDirs);
+    }
+
+    public boolean matches(Method m) {
+        String name = m.getDeclaringClass().getSimpleName() + '.' + m.getName();
+
+        if (includeGlobs.isEmpty() || includeGlobs.stream().anyMatch(f -> f.matcher(name).matches())) {
+            return skipGlobs.isEmpty() || !skipGlobs.stream().anyMatch(f -> f.matcher(name).matches());
+        }
+
+        return false;
+    }
+
+    private static Pattern filterFrom(String s) {
+        if (s.startsWith("*") && s.endsWith("*")) {
+            // contains.
+            return Pattern.compile(".*" + Pattern.quote(s.substring(1, s.length() - 1)) + ".*");
+        }
+        if (s.startsWith("*")) {
+            // ends with.
+            return Pattern.compile(".*" + Pattern.quote(s.substring(1)) + "$");
+        }
+        if (s.endsWith("*")) {
+            // starts with.
+            return Pattern.compile("^" + Pattern.quote(s.substring(0, s.length() - 1)) + ".*");
+        }
+
+        // equals
+        return Pattern.compile("^" + Pattern.quote(s) + "$");
+    }
+
+    private static List<Pattern> filterFromGlobs(List<String> globs) {
+        Set<String> result = new HashSet<>();
+        for (String g : globs) {
+            if (!g.contains(".") && !g.contains("*")) {
+                // all lowercase, folder name.
+                if (g.equals(g.toLowerCase())) {
+                    result.add(g.substring(0, 1).toUpperCase() + g.substring(1).toLowerCase() + "Tests.*");
```

Comment:
`continue;` ?

But I'd rewrite all this loop without `continue` using a plain one-level if-else chain - this will be easier to follow, IMO:
```
if (g.contains(".") || g.contains("*")) {
   ...
} else if (...) {
   ...
} else if ...
```